src/components/datepicker/js/calendar.js

```diff
@@ -197,13 +200,18 @@
 
     var boundKeyHandler = angular.bind(this, this.handleKeyEvent);
 
-    // Bind the keydown handler to the body, in order to handle cases where the focused
-    // element gets removed from the DOM and stops propagating click events.
-    angular.element(document.body).on('keydown', boundKeyHandler);
+    if (this._panelRef) {
+      // Bind the keydown handler to the body, in order to handle cases where the focused
+      // element gets removed from the DOM and stops propagating key events.
+      angular.element(document.body).on('keydown', boundKeyHandler);
```

Comment:
The calendar currently focuses the active date's cell (I believe for accessibility reasons), which causes any keyboard events to be captured by the calendar. The issue is that the calendar uses `mdVirtualRepeat` which will remove the cell eventually, if you scroll long enough. This causes the browser to shift focus back to the body which stops the keyboard events from working.